warehouse/query-core/src/main/java/datawave/query/tables/RemoteEventQueryLogic.java

```diff
@@ -0,0 +1,276 @@
+     */
+    public RemoteEventQueryLogic(RemoteEventQueryLogic other) {
+        super(other);
+        
+        if (log.isTraceEnabled())
+            log.trace("Creating Cloned RemoteQueryLogic: " + System.identityHashCode(this) + " from " + System.identityHashCode(other));
+        
+        setRemoteQueryService(other.getRemoteQueryService());
+        
+        // Set ShardQueryConfiguration variables
+        setConfig(RemoteQueryConfiguration.create(other));
+    }
+    
+    public String getRemoteId() {
+        return getConfig().getRemoteId();
+    }
+    
+    public void setRemoteId(String id) {
+        getConfig().setRemoteId(id);
+    }
+    
+    public String getRemoteQueryLogic() {
+        return getConfig().getRemoteQueryLogic();
+    }
+    
+    public void setRemoteQueryLogic(String remoteQueryLogic) {
+        getConfig().setRemoteQueryLogic(remoteQueryLogic);
+    }
+    
+    public Object getCallerObject() {
+        return getPrincipal();
+    }
+    
+    @Override
+    public GenericQueryConfiguration initialize(Connector connection, Query settings, Set<Authorizations> auths) throws Exception {
+        GenericResponse<String> createResponse = remoteQueryService.createQuery(getRemoteQueryLogic(), settings.toMap(), getCallerObject());
+        setRemoteId(createResponse.getResult());
+        return getConfig();
+    }
+    
+    @Override
+    public String getPlan(Connector connection, Query settings, Set<Authorizations> auths, boolean expandFields, boolean expandValues) throws Exception {
+        GenericResponse<String> planResponse = remoteQueryService.planQuery(getRemoteQueryLogic(), settings.toMap());
+        return planResponse.getResult();
+    }
+    
+    @Override
+    public void setupQuery(GenericQueryConfiguration genericConfig) throws Exception {
+        if (!RemoteQueryConfiguration.class.isAssignableFrom(genericConfig.getClass())) {
+            throw new QueryException("Did not receive a RemoteQueryConfiguration instance!!");
+        }
+        
+        config = (RemoteQueryConfiguration) genericConfig;
+        
+        // Create an iterator that returns a stream of EventBase objects
+        this.iterator = new RemoteQueryLogicIterator();
+    }
+    
+    private QueryLogicTransformer transformerInstance = null;
+    
+    @Override
+    public QueryLogicTransformer getTransformer(Query settings) {
+        // a transformer that turns EventBase objects into a response
+        if (this.transformerInstance != null) {
+            addConfigBasedTransformers();
+            return this.transformerInstance;
+        }
+        
+        MarkingFunctions markingFunctions = this.getMarkingFunctions();
+        EventBaseTransformer transformer = new EventBaseTransformer(settings, markingFunctions, getResponseObjectFactory());
+        
+        this.transformerInstance = transformer;
+        addConfigBasedTransformers();
+        return this.transformerInstance;
```

Comment:
could theoretically drop `this` throughout this method, seems potentially unnecessary and inconsistent with `this` usage thorughout the rest of the class.